### ISSUE TITLE
[somfycul] New Binding somfycul

### ISSUE THREAD

**Issue by Nookyx** [openhab/openhab-addons#15974]
I'm planning on adding a binding for the somfy RTS rollershutters via CUL devices (e.g. nanoCUL). 
Opening the feature request here as per contribution guidelines.

This will be based off the initial work done by Daniel Weisser back then with the [somfycul binding](https://github.com/weisserd/openhab2-addons) (he never released it officially).

I have asked the original author (Daniel Weisser) in the community forum for permissions to further develop his addon, and he [agreed ](https://community.openhab.org/t/openhab-culfw-for-somfy-rts-rollershutters/18905/165?u=nookyx) to it.

When adapting the addon, I'll also try to fix the [gnu.io dependency issue](https://github.com/openhab/openhab-addons/issues/7573).

**Comment by lsiepel** [comment]
Did you succeed in binding development?

**Comment by Nookyx** [comment]
Sorry for the late reply.
I didn't have the time to finish it, yet. However, I've continued working on it just now.
I think I can create a PR soon(tm)

The code has been updated to compile against OH5 and it seems to pass the tests.
As it is my very first contribution, there could very well be a few things that need changing in the review, but I'll do my best to follow the OH development/contribution guidelines.

**Comment by lsiepel** [comment]
This can be a usefull checklist: https://github.com/openhab/openhab-addons/wiki/Review-Checklist

The hardware for htis bindind is specific and warrants an own binding, but there seem to be 5-6 related bindings that up to some extend control these devices. Did you consider to add this binding to the marketplace or add it to the main repository?

Did you check if the functionality can be integrated into any of these:
- Somfy myLink Binding
- Somfy Tahoma Binding
- Somfy URTSI II Binding
- RFXCOM / RFLink (via third-party bindings)
- Velux KLF200
- Matter ?

**Comment by Nookyx** [comment]
My goal is to add it to the main repository.
I've looked at the existing bindings and all of them serve a specific purpose that is also primarily hardware-bound:

- Somfy myLink Binding
  - requires a Somfy myLink Hub (device) that handles communication with the shutters, comes at 300$
- Somfy Tahoma Binding
  - same as myLink, it's a Hub device (at 200$) that the vendor sells for the communication with the shutters 
- Somfy URTSI II Binding
  - comes close to what a CUL stick does, also sold as a specific device by the vendor
- RFXCOM / RFLink (via third-party bindings)
  - feels like a general purpose binding for RF communication, whereas the goal of my binding is for a specific hardware-bound scenario (controlling somfy RTS shutters via a CUL stick, e.g. attached to a raspberry pi). Somfy RTS is very specific protocol.
- Velux KLF200
  - completely different vendor, incompatible with somfy
- Matter ?
  - does not apply, this is about RTS communication with somfy shutters. Can only do 433.42 MHz RF of the Somfy RTS protocol.

**Comment by Nookyx** [comment]
PR has been created

**Comment by lsiepel** [comment]
JFTR the KLF200 does control Somfy devices. It does io-homecontrol, not sure about rts

**Comment by Nookyx** [comment]
> JFTR the KLF200 does control Somfy devices. It does io-homecontrol, not sure about rts
you're right, but only for somfy io. Somfy io is not somfy RTS though, there are major technical differences between these two protocols and they are not compatible. 

The Velux KLF200 is basically something like the Somfy Tahoma device, but from a different vendor and it happens to also support the somfy io protocol (I think somfy started using industry standards with the io protocol, which is not the case with their proprietary RTS protcol).
